scenarios/aws/microVMs/microvms/run.go

```diff
@@ -123,23 +120,23 @@ func newMetalInstance(e commonConfig.CommonEnvironment, name, arch string, m con
 	}
 
 	return &Instance{
-		e:             &InstanceEnvironment{awsEnv.CommonEnvironment, &awsEnv},
+		e:             instanceEnv,
 		instance:      awsInstance,
 		Connection:    conn.ToConnectionOutput(),
 		Arch:          arch,
 		instanceNamer: namer,
 	}, nil
 }
 
-func newInstance(e commonConfig.CommonEnvironment, arch string, m config.DDMicroVMConfig) (*Instance, error) {
-	name := e.Ctx.Stack() + "-" + arch
+func newInstance(instanceEnv *InstanceEnvironment, arch string, m config.DDMicroVMConfig) (*Instance, error) {
+	name := instanceEnv.Ctx.Stack() + "-" + arch
```

Comment:
should we check the pointer for nil?

Reply:
I think it would be fine to panic here if the pointer is nil, since a valid environment can be regarded as an invariant for this program..